container-search/src/main/java/com/yahoo/prelude/semantics/engine/Evaluation.java

```diff
@@ -260,34 +272,49 @@ public void insertItem(Item item, CompositeItem parent, int index, TermType desi
         }
 
         if (( desiredParentType == TermType.DEFAULT || desiredParentType.hasItemClass(parent.getClass()) )
-             && equalIndexNameIfParentIsPhrase(item, parent)) {
-            addItem(parent, index, item, desiredParentType);
+             && equalIndexNameIfParentIsPhrase(items, parent)) {
+            for (Item item : items)
+                addItem(parent, index, item, desiredParentType);
+        }
+        else if (parent.items().isEmpty()) {
+            CompositeItem parentsParent = parent.getParent();
+            CompositeItem newParent = newParent(desiredParentType);
+            items.forEach(item -> newParent.addItem(item));
+            parentsParent.setItem(parentsParent.getItemIndex(parent), newParent);
+        }
+        else if (items.size() ==1 && desiredParentType.hasItemClass(items.get(0).getClass())) { // This will never happen
+            for (Item item : items)
+                addItem(parent, index, item, desiredParentType);
         }
         else if (incompatible(desiredParentType, parent)) {
-            insertIncompatibleItem(item, parent, query, desiredParentType);
+            for (Item item : items)
+                insertIncompatibleItem(item, parent, query, desiredParentType);
         }
         else {
-            insertIncompatibleItemAsParent(item, parent, query, desiredParentType);
+            for (Item item : items)
+                insertIncompatibleItemAsParent(item, parent, query, desiredParentType);
         }
     }
 
+    /** Returns true if this item represents an empty query *tree*. */
     private boolean isEmpty(Item item) {
         if (item == null) return true;
         if (item instanceof QueryTree && ((QueryTree) item).isEmpty()) return true;
         return false;
     }
 
     /** Returns true if the desired type cannot have childCandidate as a child */
-    private boolean incompatible(TermType desiredParentType, Item childCandidate) {
-        return desiredParentType == TermType.EQUIV && childCandidate.getItemType() != Item.ItemType.EQUIV;
+    private boolean incompatible(TermType desiredParentType, CompositeItem parent) {
```

Comment:
changed parameter name makes the comment (even more) incomprehensible

Reply:
You are right; I did a round of cleanup now. Should hopefully address the phrase/equiv clarity issue also.